### PR TITLE
Fix eeglab reader when reading overlapping epochs with duplicated events

### PR DESCRIPTION
**Pull request by mmagnuski**
In rare cases when two consecutive epochs are overlapping (just by chance) some of the events could be duplicated across these epochs. Then `ev_idx` (see the code) would be increased by too much leading to bad latencies and sometimes also to an indexing error.

### Example 1
It's easier to explain this using an example. Let's suppose we have epochs A and B, but they are ovelapping so that one of the events (event y) is present in both epochs:
```
epoch A contains:
event x at latency 0, event y at latency 223

epoch B contains:
event y at latency 0
```
(note: the latencies above are within-epoch with 0 referring to time 0 - this is how `EEG.epoch` stores them)
Current implementation of the eeglab epoch reader does not assume duplicated events, so it just counts the occurrences of events and in this case even though there are 2 events in total the code would try to look at a third event in `EEG.event` structure. This leads to an indexing error.
I fix it by using the `urevent` field of the event which contains unique event identifiers.

### Example 2
Now lets add epoch C, with two additional events - z and q:
```
epoch A contains:
event x at latency 0, event y at latency 223

epoch B contains:
event y at latency 0

epoch C contains:
event z at latency -100, event q at latency 0
```
(note: the latencies above are within-epoch with 0 referring to time 0 - this is how `EEG.epoch` stores them)
Now, even though event `y` is duplicated, we don't get an indexing error. This is because when looking at epoch B, mne reader thinks it is seeing the third event, but we have four events in total (so no indexing error). Then when looking at epoch C, there are two events, mne reader considers only the first one for storing the `EEG.event` latency, which it assumes to be the fourth event. We have four unique events in total so still no indexing error.
However, when taking the latency from `EEG.event` in epoch B it takes the latency of event `z`, not `y` and for epoch C it takes the latency of event `g`, not `z`.
Taking the latencies of different events is not a big problem as these latencies are taken from `EEG.event` which for epoched data stores the latency of the events with respect to the epoched concatenated data (so I don't think anyone actually relies on these valuese). 

## Which latencies we should take
I also changed the code to take the latency of epoch-centering event, not the first one. This is not important to the fix so I can revert this if you prefer it stayed as it was. However I'd argue for saving the latencies of epoch-centering event (the one with respect to which the epoching was done) as this is what mne does in other cases. Additionally the latencies should be taken with respect to the raw data (`EEG.urevent.latency`), not epoched-concatenated data (`EEG.event.latency`) - as this is the convention in mne `events` array. But this is a separate issue, maybe not worth changing. :)

### CODE REVIEW

**Comment by mmagnuski** [comment]
(redo of #6051)

**Comment by agramfort** [comment]
any chance to update a test to avoid breaking this in the future?

@jona-sassenhagen @cbrnr any opinion on this one?

**Comment by mmagnuski** [comment]
@agramfort Sure, I can add a test.

File: mne/io/eeglab/eeglab.py [hunk 1]
```diff
@@ -450,25 +450,33 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
         if events is None and eeg.trials > 1:
             # first extract the events and construct an event_id dict
             event_name, event_latencies, unique_ev = list(), list(), list()
-            ev_idx = 0
             warn_multiple_events = False
             epochs = _bunchify(eeg.epoch)
             events = _bunchify(eeg.event)
+
+            # construct urevent idx -> event idx mapping
+            ur2ev = {evnt.urevent: idx for idx, evnt in enumerate(events)}
```
Comment (massich) [inline mne/io/eeglab/eeglab.py]: remind me, what does `urevent` mean?

Comment (massich) [reply mne/io/eeglab/eeglab.py]: its field in the event file. ok.

**Comment by massich** [comment]
yes, we should add a test

File: mne/io/eeglab/eeglab.py (hunk 1 above)
Comment (mmagnuski) [reply mne/io/eeglab/eeglab.py]: yes, it is a field containing unique event identifier.

**Comment by mmagnuski** [comment]
> Is not the events that overlap, but the epochs.
Yes, sorry for the confusing typo in the commit message.

**Comment by massich** [comment]
@mmagnuski once you update whats_new I'm +1

**Comment by mmagnuski** [comment]
@massich No, I'll add a test before that. :) Sorry for delays, I'll try to find some time today or tomorrow in the evening.

**Comment by massich** [comment]
Woo, I forgot the **test**. YES please add them.

**Comment by mmagnuski** [comment]
I have to put this on hold because it seems it might have been error on my part - by default eeglab epoching duplicates events both in `EEG.epoch` and `EEG.event`. I have an old, faster version of eeglab epoching and it seems not to duplicate the `EEG.event` element, which leads to the error when reading epochs in mne. So the behavior I describe in the first post here does not seem to be the default of eeglab.

**Pull request closed without merging**